### PR TITLE
ENH: update scipy.optimize

### PR DESCRIPTION
**Pull request by mdhaber**

### CODE REVIEW

File: intro/scipy/index.rst [hunk 1]
```diff
@@ -278,18 +278,70 @@ interpolation example.
...
+
+    >>> def f(x):
+    ...     return (x-1)*(x-2)
+    >>> res = sp.optimize.root_scalar(f, x0=0)  # x0 is the guess
+    >>> res
+      converged: True
+           flag: converged
+ function_calls: 12
+     iterations: 6
+           root: [ 1.000e+00]
```
Comment (mdhaber) [inline intro/scipy/index.rst:301]:
```suggestion
         converged: True
              flag: converged
    function_calls: 12
        iterations: 6
              root: [ 1.000e+00]
```

File: intro/scipy/index.rst [hunk 2]
```diff
@@ -278,18 +278,70 @@ interpolation example.
...
+
+Note that only one the root at ``1.0`` is found. By inspection, we can tell
+that there is a second root at ``2.0``. We can direct the function toward a
+particular root by changing the guess or by passing a bracket that contains
```
Comment (mdhaber) [inline intro/scipy/index.rst:305]: I would like to show that changing the guess to a point where the derivative is 0 can cause problems with the gradient-based methods. Would that be too much?

**Review by mdhaber — COMMENTED** [review]
Edits

**Comment by mdhaber** [comment]
Test is failing because it needs scipy/scipy#17691. What should we do - specify the derivative so that `method='newton'` can be used in old SciPy, or wait until SciPy 1.11 comes out before merging this PR?
I'd prefer to just add the derivative temporarily, then we can remove it when SciPy 1.11.0 is the minimum supported version of the lecture notes.

**Comment by jarrodmillman** [comment]
> What should we do - specify the derivative so that `method='newton'` can be used in old SciPy, or wait until SciPy 1.11 comes out before merging this PR?

It looks like you specified the derivative in f0dd780. Would you remove that after  SciPy 1.11 is released?

**Review by jarrodmillman — APPROVED** [review]

**Comment by mdhaber** [comment]
Yes

**Pull request merged**